pkg/machine/compression/generic.go

```diff
@@ -63,3 +63,13 @@ func (d *genericDecompressor) close() {
 		logrus.Errorf("Unable to close compressed file: %q", err)
 	}
 }
+
+func (d *genericDecompressor) sparseOptimizedCopy(w WriteSeekCloser, r io.Reader) error {
+	var err error
+	sparseWriter := NewSparseWriter(w)
+	defer func() {
+		err = sparseWriter.Close()
```

Comment:
This overwrites the error from `io.Copy`, if any.

Reply:
👍 updated it